Rust type error when building docker image using Mac ARM

I'm getting rust type errors when trying to build the docker file currently.

This kind of stuff:

22.20 error[E0308]: mismatched types
22.20     --> c/src/metta.rs:1892:13
22.20      |
22.20 1888 |         let bytes_written = (api.path_for_name)(
22.20      |                             ------------------- arguments to this function are incorrect
22.20 ...
22.20 1892 |             buffer.as_mut_ptr(),
22.20      |             ^^^^^^^^^^^^^^^^^^^ expected `*mut u8`, found `*mut i8`
22.20      |
22.20      = note: expected raw pointer `*mut u8`
22.20                 found raw pointer `*mut i8`
22.20 
22.20 error[E0308]: mismatched types
22.20     --> c/src/metta.rs:1896:44
22.20      |
22.20 1896 |             vec![PathBuf::from(cstr_as_str(buffer[0..=bytes_written].as_ptr()))]
22.20      |                                ----------- ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `*const u8`, found `*const i8`
22.20      |                                |
22.20      |                                arguments to this function are incorrect
22.20      |
22.20      = note: expected raw pointer `*const u8`
22.20                 found raw pointer `*const i8`
22.20 note: function defined here
22.20     --> c/src/util.rs:8:8
22.20      |
22.20 8    | pub fn cstr_as_str<'a>(s: *const c_char) -> &'a str {
22.20      |        ^^^^^^^^^^^     ----------------
22.20 
22.20 For more information about this error, try `rustc --explain E0308`.
22.20 error: could not compile `hyperonc` (lib) due to 2 previous errors
22.20 make[5]: *** [CMakeFiles/build-hyperonc.dir/build.make:70: CMakeFiles/build-hyperonc] Error 101
22.20 make[4]: *** [CMakeFiles/Makefile2:164: CMakeFiles/build-hyperonc.dir/all] Error 2
22.20 make[3]: *** [Makefile:146: all] Error 2
22.20 make[2]: *** [CMakeFiles/hyperonc.dir/build.make:86: hyperonc-prefix/src/hyperonc-stamp/hyperonc-build] Error 2
22.20 make[1]: *** [CMakeFiles/Makefile2:86: CMakeFiles/hyperonc.dir/all] Error 2
22.21 make: *** [Makefile:101: all] Error 2
------
context:35
--------------------
  33 |     WORKDIR ${HOME}/hyperon-experimental/build
  34 |     RUN cmake ..
  35 | >>> RUN make
  36 |     RUN make check
  37 |     
--------------------
ERROR: failed to solve: process "/bin/sh -c make" did not complete successfully: exit code: 2

--- a/c/src/metta.rs
+++ b/c/src/metta.rs
@@ -1883,7 +1883,7 @@ impl FsModuleFormat for CFsModFmtLoader {
         let parent_dir_c_string = str_as_cstr(parent_dir.to_str().unwrap());
         let mod_name_c_string = str_as_cstr(mod_name);
         const BUF_SIZE: usize = 512;
-        let mut buffer = [0i8; BUF_SIZE];
+        let mut buffer = [0 as c_char; BUF_SIZE];
 
         let bytes_written = (api.path_for_name)(
             self.payload,

